File: src/main/java/br/com/leonardoferreira/jirareport/repository/HolidayRepository.java
package br.com.leonardoferreira.jirareport.repository;

import br.com.leonardoferreira.jirareport.domain.Holiday;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface HolidayRepository extends CrudRepository<Holiday, Long> {

    List<Holiday> findAllByBoardId(Long id);

    Page<Holiday> findAllByBoardId(Long id, Pageable pageable);

}

File: database-library/src/main/kotlin/br/com/jiratorio/repository/HolidayRepository.kt
package br.com.jiratorio.repository

import br.com.jiratorio.domain.entity.BoardEntity
import br.com.jiratorio.domain.entity.HolidayEntity

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface HolidayRepository : CrudRepository<HolidayEntity, Long> {

    fun findAllByBoardId(id: Long): List<HolidayEntity>

    fun findAllByBoardId(id: Long, pageable: Pageable): Page<HolidayEntity>

    fun findAllByBoard(board: BoardEntity): List<HolidayEntity>

    fun findByDateAndBoardId(date: LocalDate, boardId: Long): HolidayEntity?

    fun findByIdAndBoard(holidayId: Long, board: BoardEntity): HolidayEntity?

    fun findByIdOrNull(id: Long): HolidayEntity? =
        findById(id).orElse(null)

}
